protoc-gen-go_cli: issue with generated oneof field

This codebase doesn't compile because of an apparent bug in the gen go_cli codebase:

```
cmd/secretsctl/create-secret.go:29:49: undefined: timestamp.Secret_ExpireTime
cmd/secretsctl/create-secret.go:31:42: undefined: duration.Secret_Ttl
cmd/secretsctl/update-secret.go:31:49: undefined: timestamp.Secret_ExpireTime
cmd/secretsctl/update-secret.go:33:42: undefined: duration.Secret_Ttl
```

The proto code in question is:
https://github.com/googleapis/googleapis/blob/master/google/cloud/secretmanager/v1/resources.proto#L81

https://github.com/tmc/secretsctl/issues/2

--- a/internal/gencli/flag.go
+++ b/internal/gencli/flag.go
@@ -36,6 +36,7 @@ type Flag struct {
 	MessageImport pbinfo.ImportSpec
 	OneOfs        map[string]*Flag
 	OneOfSelector string
+	OneOfDesc     *desc.OneOfDescriptor
 	VarName       string
 	FieldName     string
 	SliceAccessor string

--- a/internal/gencli/gencli.go
+++ b/internal/gencli/gencli.go
@@ -248,7 +248,7 @@ func (g *gcli) genCommands() {
 					})
 				}
 
-				pkg, err := g.getImport(msg)
+				pkg, err := getImport(msg)
 				if err != nil {
 					continue
 				}
@@ -339,6 +339,7 @@ func (g *gcli) buildOneOfFlag(cmd *Command, msg *desc.MessageDescriptor, field *
 		IsOneOfField:  true,
 		IsNested:      isNested,
 		OneOfSelector: prefix + oneOfField,
+		OneOfDesc:     field.GetOneOf(),
 		Usage:         toShortUsage(sanitizeComment(field.GetSourceInfo().GetLeadingComments())),
 		VarName:       cmd.InputMessageVar + dotToCamel(title(oneOfPrefix+field.GetName())),
 	}
@@ -364,7 +365,7 @@ func (g *gcli) buildOneOfFlag(cmd *Command, msg *desc.MessageDescriptor, field *
 	flag.FieldName = fieldName
 
 	// construct oneof gRPC struct type info
-	parent, err := g.getImport(msg)
+	parent, err := getImport(msg)
 	if err != nil {
 		return
 	}
@@ -620,7 +621,7 @@ func (g *gcli) getFieldBehavior(field *desc.FieldDescriptor) (output bool, requi
 	return
 }
 
-func (g *gcli) getImport(m desc.Descriptor) (*pbinfo.ImportSpec, error) {
+func getImport(m desc.Descriptor) (*pbinfo.ImportSpec, error) {
 	pkg := m.GetFile().GetFileOptions().GetGoPackage()
 	appendpb := func(n string) string {
 		if !strings.HasSuffix(n, "pb") {
@@ -654,7 +655,7 @@ func (g *gcli) getImport(m desc.Descriptor) (*pbinfo.ImportSpec, error) {
 }
 
 func (g *gcli) addImport(cmd *Command, m desc.Descriptor) (*pbinfo.ImportSpec, error) {
-	pkg, err := g.getImport(m)
+	pkg, err := getImport(m)
 	if err != nil {
 		return nil, err
 	}

--- a/internal/gencli/util.go
+++ b/internal/gencli/util.go
@@ -102,7 +102,20 @@ func oneofTypeName(field, inputMsgType string, flag *Flag) string {
 	tname := fmt.Sprintf("%s_%s", inputMsgType, upperField)
 
 	if flag.IsNested {
-		tname = fmt.Sprintf("%s.%s_%s", flag.MessageImport.Name, flag.Message, upperField)
+		imp := flag.MessageImport.Name
+
+		// If the field is a message not defined in the same package as the
+		// parent, we must use the parent's import name for the oneof wrapper
+		// type.
+		if flag.IsMessage() && flag.MsgDesc.GetFile().GetPackage() != flag.OneOfDesc.GetOwner().GetFile().GetPackage() {
+			// There is no good way to handle an error from a template helper
+			// function so we will let the generation fail because the import
+			// will be empty - but this shouldn't ever happen because the Go pkg
+			// option is required.
+			p, _ := getImport(flag.OneOfDesc.GetOwner())
+			imp = p.Name
+		}
+		tname = fmt.Sprintf("%s.%s_%s", imp, flag.Message, upperField)
 	}
 
 	if flag.IsMessage() {

